src/main/java/org/prebid/server/bidder/yeahmobi/YeahmobiBidder.java

```diff
@@ -90,32 +90,35 @@ private ExtImpYeahmobi parseImpExt(Imp imp) {
         try {
             return mapper.mapper().convertValue(imp.getExt(), YEAHMOBI_EXT_TYPE_REFERENCE).getBidder();
         } catch (IllegalArgumentException e) {
-            throw new PreBidException(e.getMessage(), e);
+            throw new PreBidException(String.format("Impression id=%s, has invalid Ext", imp.getId()));
         }
     }
 
-    @SneakyThrows
     private Imp processImp(Imp imp) {
         final Native xNative = imp.getXNative();
 
         if (xNative != null) {
-            final JsonNode nativeRequest = xNative.getRequest() != null
-                    ? mapper.mapper().readValue(xNative.getRequest(), JsonNode.class)
-                    : null;
-
-            final String newNativeRequest;
-            final ObjectNode objectNode = mapper.mapper().createObjectNode().set("native", nativeRequest);
-            newNativeRequest = nativeRequest == null || nativeRequest.get("native") == null
-                    ? mapper.mapper().writeValueAsString(objectNode)
-                    : null;
-
-            return newNativeRequest != null
-                    ? imp.toBuilder().xNative(Native.builder().request(newNativeRequest).build()).build()
+            final String resolvedNativeRequest = resolveNativeRequest(xNative.getRequest());
+            return resolvedNativeRequest != null
+                    ? imp.toBuilder().xNative(Native.builder().request(resolvedNativeRequest).build()).build()
                     : imp;
         }
         return imp;
     }
 
+    @SneakyThrows
```

Comment:
Pls specify exception explicitly `@SneakyThrows(JsonProcessingException.class)` for more control, but personally i'd use just regular try-catch instead. Up to you.

Reply:
Thx, less magic - good project.